### ISSUE TITLE
Doesn't work on Monterey.

### ISSUE THREAD

**Issue by xAdamQ** [artginzburg/MiddleClick-Ventura#32]
I installed it from brew as in the readme and it have the accessibility permissions. But It doesn't work. However I installed it from a random source on the internet later and it worked but it doesn't open at startup after putting it in login items and closes itself after a while.

**Comment by Cediddi** [comment]
It was not working for me as well, then I tried to run it via terminal `/Applications/MiddleClick.app/Contents/MacOS/MiddleClick` and it's working flawlessly. I think there's a configuration error or something.

**Comment by xAdamQ** [comment]
Thanks a lot, it actually worked that way. To launch it at startup:

Start Automator.app;
Select "Application";
Click "Show library" in the toolbar (if hidden);
Add "Run shell script" (from the Actions/Utilities);
Copy-and-paste your script into the window;
Save it somewhere: a file called your_name.app will be created);
Close the middle click and run this the created app to test
Go to System Preferences → Users and Groups → Login items
Add this newly-created app;

[source](https://superuser.com/questions/229773/run-command-on-startup-login-mac-os-x)

**Comment by 2mawi2** [comment]
I am getting 
`/Applications/MiddleClick.app/Contents/MacOS/MiddleClick
*** Recognized (0x69) family*** (30 cols X 22 rows)
2021-12-20 21:27:58.719 MiddleClick[73964:416759] Couldn't create event tap! Check accessibility permissions.
2021-12-20 21:28:03.724 MiddleClick[73964:416759] Restarting app...`  even though my privacy accessibility settings are configured correctly. I debugged into the code and it seems like the event tap does never gets created. Does somebody know how to fix this ?

**Comment by leonard84** [comment]
I can confirm that the workaround through the shell works, but it has the annoying side effect that the launching application iterm/automator gets an accessibility feature prompt. If you deny the request it still works, but I'd like to avoid having to grant them the permission just to start this app without dialogs.

**Comment by chrisma** [comment]
Had the same issue of MiddleClick not working after brew installation. #24 mentioned activating MiddleClick in the 'Accessibility' section of the 'Security & Privacy' Preferences )it was not activated after install). This worked for me on Monterey 12.1.

![image](https://user-images.githubusercontent.com/1652117/149638080-f9cce45f-3f3d-429d-afff-be25c3362c20.png)

**Comment by JarrettBillingsley** [comment]
I'm on monterey and trying to find a replacement for magicprefs (😭) and I tried this out. It's very inconsistent. It seems to turn about 5% of my three-finger clicks into middle clicks and misses the rest. 

Man, why's apple gotta make this such a pain?

**Comment by JarrettBillingsley** [comment]
Something possibly diagnostic about why this doesn't work (I have no idea, I'm just looking at the source):

https://github.com/artginzburg/MiddleClick-BigSur/blob/be5ee82bdfdc07a29cb45b901478b56ae1ce832e/MiddleClick/Controller.m#L181

This function is hardcoded to check for `kCGEventLeftMouseDown` and `kCGEventLeftMouseUp`, but I notice that a lot of the time it fails to middle click, I get a right-click menu instead. So maybe this code is just not being triggered because with three fingers down, mac OS is interpreting that as a right-click. Maybe this changed at some point and this code needs to be updated?

**Comment by chrisma** [comment]
> trying to find a replacement for magicprefs

Have you tried BetterTouchTool yet? There seem to be [some reports](https://www.reddit.com/r/BetterTouchTool/comments/3gv5j3/open_link_in_new_tab_safari/) of that working

**Comment by JarrettBillingsley** [comment]
I haven't, but I really only need the middle click, so something like that feels a little overkill. But I imagine it will have the same limitations because...

I got a response from the developer of Middle and it turns out the API that all these apps use is limited. Apple's code does "resting touch rejection" which filters out certain touches at the edges of the trackpad, which explains why it's picky about where I put my fingers. I don't know that any of these apps can fix that issue without resorting to weird hacks.

So basically, I guess I have to retrain my fingers.

**Comment by todivefor** [comment]
I used Better Touch Tool just for this. It is overkill. When I got my Macbook Air M1 and Monterey, I tried this. It works fine. I use pointer, middle, ring fingers and it works 100%. I have to restart occasionally after sleep, but I can live with that. I couldn't get 2 finger tap to work, but now that I am using 3, I am used to it.

**Comment by Filaipus** [comment]
Yea there's something wrong going on. I can only get this to work properly in Monterey if I use three fingers tap. And I had to restart a bunch of times, leading to a few hard freezes from trying to add/remove the permissions in the accessibility options.

**Comment by amjd** [comment]
I've tried MiddleClick, Middle and BetterTouchTool and none of them seem to work properly on Monterey for me.

**MiddleClick:** Keeps restarting every few seconds. If you run it via terminal this is what is logged:
```
*** Recognized (0xb0) family*** (60 cols X  2 rows)
*** Recognized (0x6b) family*** (26 cols X 18 rows)
2022-05-18 14:13:57.786 MiddleClick[25570:560430] Couldn't create event tap! Check accessibility permissions.
2022-05-18 14:14:02.791 MiddleClick[25570:560430] Restarting app...
```
This is with accessibility permission already granted.

**Middle:** Quits on sleep.

**BetterTouchTool:** Sometimes doesn't recognize the gesture.

Please let me know if anyone knows other workarounds or apps.

**Comment by byprakoso** [comment]
Still doesn't work in monterey for me

**Comment by amjd** [comment]
Update after two months: I've found the perfect solution in BetterTouchTool. I gave it another shot and it's working reliably this time.

**Comment by Jont828** [comment]
Although it works well it is worth noting that it costs $22 whereas MiddleClick is free and open source. While I'm not too familiar with writing Mac apps, I'll look into it if I have time.

**Comment by JarrettBillingsley** [comment]
For what it's worth, if you only need middle-click functionality, the Middle app is currently only $8, almost a third the price. I would love there to be a functional open-source program for this, but I just got Middle cause I was tired of this app breaking.
